### PR TITLE
NEW : allow stock management by product

### PR DESCRIPTION
**Pull request by thomas-Ngr**
# NEW : allow stock management by product

When using product stocks, users may want to have some products for which they don't want to manage stocks, for example in case of subcontracting.

This PR:
- adds a field "Stockable product" on product card (editable on update, not on create)
- sets this option to 1 by default
- hides the "stocks" tabs if this option is not set
- does not manage stock movements for a product if this option is not set.

Discussion started on #26184 and #26200

SQL Part : #26190

### CODE REVIEW

**Comment by eldy** [comment]
Change into language file  htdocs/langs/fr_FR/products.lang   is not allowed  (only en_US is allowed for language files)

File: htdocs/expedition/card.php [hunk 1]
```diff
@@ -343,6 +344,30 @@
...
 			} else {
+				$p = new Product($db);
+				$res = $p->fetch($objectsrc->lines[$i]->fk_product);
+				if ($res > 0) {
+					if (GETPOST('entrepot_id', 'int') == -1) {
+						$qty .= '_'.$j;
+					}
+
+					if ($p->stockable_product == Product::DISABLED_STOCK) {
+						$w = new Entrepot($db);
```
Comment (eldy) [inline htdocs/expedition/card.php:355]: If product is not stockable, we have no reason to manage some warehouses.

File: htdocs/expedition/card.php [hunk 2]
```diff
@@ -343,6 +344,30 @@
...
+						$Tw = $w->list_array();
+						if (count($Tw) > 0) {
+							$w_Id = array_keys($Tw);
+							$stockLine[$i][$j]['qty'] = GETPOST($qty, 'int');
+
+							// lorsque que l'on a le stock désactivé sur un produit/service
```
Comment (eldy) [inline htdocs/expedition/card.php:361]: English comment not allowed

File: htdocs/expedition/class/expedition.class.php [hunk 3]
```diff
@@ -2829,6 +2833,13 @@ class ExpeditionLigne extends CommonObjectLine
 	public $volume;
 	public $volume_units;
 
+	/**
+	 * 0=This service or product is not managed in stock, 1=This service or product is managed in stock
+	 *
+	 * @var boolean
```
Comment (eldy) [inline htdocs/expedition/class/expedition.class.php:2839]: @var int

File: htdocs/expedition/class/expedition.class.php [hunk 4]
```diff
@@ -2829,6 +2833,13 @@ class ExpeditionLigne extends CommonObjectLine
...
+	 */
+	public $stockable_product = true;
```
Comment (eldy) [inline htdocs/expedition/class/expedition.class.php:2841]: public $stockable_product = 1;

**Comment by thomas-Ngr** [comment]
@eldy I guess all threads can be resolved, and the checks pass. Can you have another look at this PR ? Thanks :+1:

**Comment by thomas-Ngr** [comment]
closing and restarting from branch Develop since file expedition.class.php has been split and there is no easy way to resolve conflicts. see #30996

**Pull request closed without merging**